sdk/eventhub/event-hubs/test/internal/auth.spec.ts

```diff
@@ -11,172 +11,188 @@ import { EnvVarKeys, getEnvVars } from "../public/utils/testUtils";
 import chai from "chai";
 import { AzureNamedKeyCredential, AzureSASCredential } from "@azure/core-auth";
 import { createSasTokenProvider } from "@azure/core-amqp";
+import { testWithServiceTypes } from "../public/utils/testWithServiceTypes";
+import { createMockServer } from "../public/utils/mockService";
 
 const should = chai.should();
-const env = getEnvVars();
-
-describe("Authentication via", () => {
-  const {
-    endpoint,
-    fullyQualifiedNamespace,
-    sharedAccessKey,
-    sharedAccessKeyName
-  } = parseEventHubConnectionString(env[EnvVarKeys.EVENTHUB_CONNECTION_STRING]);
-  const service = {
-    connectionString: env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-    path: env[EnvVarKeys.EVENTHUB_NAME],
-    endpoint: endpoint.replace(/\/+$/, "")
-  };
-
-  before(() => {
-    should.exist(
-      env[EnvVarKeys.EVENTHUB_CONNECTION_STRING],
-      "define EVENTHUB_CONNECTION_STRING in your environment before running integration tests."
-    );
-    should.exist(
-      env[EnvVarKeys.EVENTHUB_NAME],
-      "define EVENTHUB_NAME in your environment before running integration tests."
-    );
-  });
-
-  describe("Keys", () => {
-    describe("using connection string", () => {
-      it("EventHubConsumerClient", async () => {
-        const consumerClient = new EventHubConsumerClient(
-          "$Default",
-          service.connectionString,
-          service.path
-        );
-
-        const properties = await consumerClient.getEventHubProperties();
-        should.exist(properties);
-
-        await consumerClient.close();
-      });
 
-      it("EventHubProducerClient", async () => {
-        const producerClient = new EventHubProducerClient(service.connectionString, service.path);
-
-        const properties = await producerClient.getEventHubProperties();
-        should.exist(properties);
-
-        await producerClient.close();
-      });
+testWithServiceTypes("internal/auth.spec.ts", (serviceVersion) => {
```

Comment:
why do you need the file paths?

Reply:
Strictly speaking, they aren't needed. However I found it made it really easy to find out where a test that fails lives.

Reply:
[NIT] Test objects/blocks do have the file path in them, which is what I used to locate the recordings folder in the recorder.
Since you have the wrapper and using describe blocks, I'm wondering if you can automate that instead of writing in each of the files.

Reply:
Do you mean there's an API to get the file path? Does it also work for browsers?

Reply:
Yes.

Something like this.
![image](https://user-images.githubusercontent.com/10452642/133507688-ebf74526-ac1e-4ae7-b274-78ec6041e544.png)

Reply:
![image](https://user-images.githubusercontent.com/10452642/133508104-87561dcd-773b-4bb9-aaba-d428e6f885ba.png)

Reply:
Hmm, this does mean I need to wrap my `describe(this.file, ...)` in another `describe` so that I have access to the mocha suite context object when defining the describe that has the file path. I decided to go ahead and do that.

Unfortunately, that does not seem to work in browsers, but I'll accept that.

Reply:
Interesting